pyvista/plotting/utilities/sphinx_gallery.py

```diff
@@ -110,6 +119,10 @@ class DynamicScraper:
 
     """
 
+    def __repr__(self):
+        """Return a stable representation of the class instance."""
+        return f"<{type(self).__name__} object>"
+
```

Comment:
@alexrockhill, why was this needed? We do not intend for this class to be used or displayed to users as far as I understand? 

Asking because I'm going to see if I can re-land this PR with less impact

Reply:
It was added when I was trying to add coverage of the dynamic scraper but then not removed when adding that coverage exposed a bug where the dynamic scraper failed to close plots and so that was punted to a future PR.